Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM nvidia/cuda:11.8.0-cudnn8-runtime-ubuntu22.04
```

Comment:
hello @seungwoo-ji-03 
Thanks for the contribution but I do have a question regarding the image used in the dockerfile. Can you elaborate the reason for choosing GPU based image in dockerfile?

Reply:
Thank you for your question @rlrahulkanojia 

Thank you for your response!!

The reason I chose that base image is because I found packages like [nvidia-cublas-cu12](https://github.com/SylphAI-Inc/AdalFlow/blob/e7357a025ed496586350b932720ba0752fbf7324/poetry.lock#L2921) and [nvidia-cuda-cupti-cu12](https://github.com/SylphAI-Inc/AdalFlow/blob/e7357a025ed496586350b932720ba0752fbf7324/poetry.lock#L2932) in the poetry.lock file.

Additionally, since it is specified in the [pyproject.toml file that PyTorch version 2.0 or higher](https://github.com/SylphAI-Inc/AdalFlow/blob/e7357a025ed496586350b932720ba0752fbf7324/pyproject.toml#L41) should be installed, I chose CUDA 11.8, which supports all versions from PyTorch 2.0 to the latest version.

I would appreciate any additional thoughts or feedback you may have!

Reply:
Having a default GPU based docker image is not a good practice for  Non GPU Users (Mac) users as the Docker Image size is ~5GB after being compiled from the given dockerfile which is too much. 
Since Cuda base image is not valid for mac, after building it myself, the required docker image size is only 1.5 GB

Reply:
I completely agree with what you mentioned.
I reduced the size significantly (to below 2GB) by switching the base Docker image to a lighter one.

Additionally, I built the image myself and ran various tutorial scripts, all of which worked without any issues.